### PR TITLE
Update Petfinder API

### PR DESCRIPTION
**Pull request by mloberg**
Petfinder released a new version of their API. Update info for new version

Thank you for taking the time to work on a Pull Request for this project!

To ensure your PR is dealt with swiftly please check the following:

- [X] Your submissions are formatted according to the guidelines in the [contributing guide](CONTRIBUTING.md)
- [X] Your additions are ordered alphabetically
- [X] Your submission has a useful description
- [X] The description does not end with punctuation
- [X] Each table column should be padded with one space on either side
- [X] You have searched the repository for any relevant issues or pull requests
- [X] Any category you are creating has the minimum requirement of 3 items
- [X] All changes have been [squashed][squash-link] into a single commit

[squash-link]: <https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit>

### CODE REVIEW

**Comment by mloberg** [comment]
I'm not sure why the Petfinder API link is 404ing from Travis. Seems to be working for me. https://www.petfinder.com/developers/v2/docs/

**Comment by riderjensen** [comment]
Looking around, it looks like the old link is not working because instead of [https://www.petfinder.com/developers/api-docs](https://www.petfinder.com/developers/api-docs) which works, it is listed as [https://www.petfinder.com/developers/api-docs/](https://www.petfinder.com/developers/api-docs/). The old docs are being phased out by January 2020 but since Travis is 404ing on the new docs, I am going to open a PR so at least the link is working to the old docs since I cant edit this one. There is also a note on the older docs which will point people to the new docs.

**Comment by riderjensen** [comment]
Seems like Travis is 404 on the old docs page too, see https://github.com/toddmotto/public-apis/pull/926

File: README.md [hunk 1]
```diff
@@ -324,7 +324,7 @@ API | Description | Auth | HTTPS | CORS |
 | [RoutingNumbers.info](https://www.routingnumbers.info/api/index.html) | ACH/NACHA Bank Routing Numbers | No | Yes | Unknown |
 | [Tradier](https://developer.tradier.com) | US equity/option market data (delayed, intraday, historical) | `OAuth` | Yes | Yes |
 | [VAT Rates](https://jsonvat.com/) | A collection of all VAT rates for EU countries | No | Yes | Unknown |
-| [YNAB](https://api.youneedabudget.com/) | Budgeting & Planning | `OAuth` | Yes | Yes | 
+| [YNAB](https://api.youneedabudget.com/) | Budgeting & Planning | `OAuth` | Yes | Yes |
```
Comment (yannbertrand) [inline README.md]: Can you add this space back please?

Comment (mloberg) [reply README.md]: done

**Comment by yannbertrand** [comment]
Thank you!

**Pull request merged**